test/skate_web/controllers/detours_controller_test.exs

```diff
@@ -224,64 +224,66 @@ defmodule SkateWeb.DetoursControllerTest do
              } = json_response(conn, 200)
     end
 
+    @tag :authenticated
+    test "input snapshot matches the retrieved serialized detour", %{conn: conn} do
+      detour_id = 4
+
+      detour_snapshot =
+        :detour_snapshot
+        |> build()
+        |> with_id(detour_id)
+
+      put(conn, "/api/detours/update_snapshot", %{"snapshot" => detour_snapshot})
+
+      conn = get(conn, "/api/detours/#{detour_id}")
+
+      assert detour_snapshot == json_response(conn, 200)["data"]["state"]
+    end
+
     @tag :authenticated
     test "log an error if the serialized detour does not match db state", %{conn: conn} do
       detour_id = 4
 
-      snapshot = %{
-        "context" => %{
-          "routePattern" => %{
-            "headsign" => "Headsign",
-            "directionId" => 0,
-            "route" => %{
-              "name" => "23"
-            }
-          },
-          "nearestIntersection" => "Street A & Avenue B",
-          "uuid" => detour_id
-        }
-      }
+      detour_snapshot =
+        :detour_snapshot
+        |> build()
+        |> with_id(detour_id)
 
-      put(conn, "/api/detours/update_snapshot", %{"snapshot" => snapshot})
+      put(conn, "/api/detours/update_snapshot", %{"snapshot" => detour_snapshot})
+
+      retrieved_detour = Detours.get_detour!(detour_id)
+      # Changing the status is a sure way to force a fallback, as it should always be "active"
+      edited_snapshot = put_in(detour_snapshot["status"], nil)
+      edited_detour = %{retrieved_detour | state: edited_snapshot}
 
       log =
         CaptureLog.capture_log(fn ->
-          get(conn, "/api/detours/#{detour_id}")
+          Skate.Detours.SnapshotSerde.serialize(edited_detour)
```

Comment:
Sorry, could you do this via `get` instead of calling `Skate.Detours.SnapshotSerde.serialize` directly?

Same here [`99c5e17` (#2899)](https://github.com/mbta/skate/pull/2899/commits/99c5e172d003962725f1e6c8674d64992c4a8434#diff-6b5f7db841fd66bb4a09b0870661a72d957ad6cf263cccf090ebb4a251bc0687R286)

It's a _little_ strange to have a test in our controller tests that doesn't call the `get api/detours/:id` endpoint, but I do like that these tests _are_ done in this suite and I don't think it's worth pulling it out into a test suite for `         Skate.Detours.SnapshotSerde` specifically.

Reply:
Ok, this is what I was struggling with. When do I edit the detour into something that is invalid? If I add to the db via put, and retrieve from the db via get, I have no opportunity to create `edited_detour` to force the logged error

Reply:
Something like this?
```diff
Commit ID: 2974bc649c2d65fba1acb1062e066c0a31cb6898
Change ID: ytznlvkvoxysnvtrvllxpspkmkrnyotr
Author: Kayla Firestack <kfirestack@mbta.com> (2024-11-19 09:23:14)
Committer: Kayla Firestack <kfirestack@mbta.com> (2024-11-19 09:26:19)

    (no description set)

diff --git a/test/skate_web/controllers/detours_controller_test.exs b/test/skate_web/controllers/detours_controller_test.exs
index 7ad3f7169f..0c02ce4b30 100644
--- a/test/skate_web/controllers/detours_controller_test.exs
+++ b/test/skate_web/controllers/detours_controller_test.exs
@@ -252,16 +252,17 @@
         |> build()
         |> with_id(detour_id)
 
-      put(conn, "/api/detours/update_snapshot", %{"snapshot" => detour_snapshot})
+      conn = put(conn, "/api/detours/update_snapshot", %{"snapshot" => detour_snapshot})
 
-      retrieved_detour = Detours.get_detour!(detour_id)
       # Changing the status is a sure way to force a fallback, as it should always be "active"
-      edited_snapshot = put_in(detour_snapshot["status"], nil)
-      edited_detour = %{retrieved_detour | state: edited_snapshot}
+      detour_id
+      |> Detours.get_detour!()
+      |> Skate.Detours.Detours.change_detour(%{state: put_in(detour_snapshot["status"], nil)})
+      |> Skate.Repo.update()
 
       log =
         CaptureLog.capture_log(fn ->
-          Skate.Detours.SnapshotSerde.serialize(edited_detour)
+          get(conn, "/api/detours/#{detour_id}")
         end)
 
       assert log =~
```

Reply:
Ohhhh I see, a re-insert to the db with edited state only. Yeah, can do!